process/data_processor.py

```diff
@@ -0,0 +1,155 @@
+"""
+loads and preprocesses the structured log data for anomoly prediction
+"""
+import numpy as np
+import pandas as pd
+import time
+import re
+from collections import OrderedDict
+from collections import Counter
+from scipy.special import expit
+
+
+def collect_event_ids(data_frame):
+    """
+    turns input data_frame into a 2 columned dataframe
+    with columns: BlockId, EventSequence
+    where EventSequence is a list of the events that happened to the block
+    """
+    data_dict = OrderedDict()
+    for _, row in data_frame.iterrows():
+        blk_id_list = re.findall(r"(blk_-?\d+)", row["Content"])
```

Comment:
the regex and column labels should be inputs to the function to make it more generic

Reply:
Can you think of any use cases?
I cant think of why we might want to change the regex or the column names

Im pretty against pre optimizing things like this

Reply:
if we end up trying it on another file type (as was suggested in the lab), doubt we'll get there but should probably at least make it generalizable to do so?

Reply:
still feels a bit early

but it has been changed